### PR TITLE
Add mutation to update AssetImage fields

### PR DESCRIPTION
**Pull request by solocommand**
```gql
mutation UpdateAssetImage($image: UpdateAssetImageMutationInput!){
  updateAssetImage(input:$image){
    id
    isLogo
    src
  }
}
```
```json
{
  "image": {
    "id": "5defb1f4a367b8ac008b45da",
    "payload": {
      "isLogo": true
    }
  }
}
```

### CODE REVIEW

File: services/graphql-server/src/graphql/definitions/platform/asset.js [hunk 1]
```diff
@@ -6,6 +6,10 @@ extend type Query {
   assetImage(input: AssetImageQueryInput!): AssetImage @findOne(model: "platform.Asset", using: { id: "_id" }, criteria: "assetImage")
 }
 
+extend type Mutation {
+  assetImageUpdate(input: AssetImageMutationInput!): AssetImage!
```
Comment (zarathustra323) [inline services/graphql-server/src/graphql/definitions/platform/asset.js]: updateAssetImage please. 🙂

File: services/graphql-server/src/graphql/resolvers/platform/asset.js [hunk 2]
```diff
@@ -14,4 +18,24 @@ module.exports = {
     alt: image => createAltFor(image),
     caption: image => createCaptionFor(image.caption),
   },
+
+  /**
+   *
+   */
+  Mutation: {
+    assetImageUpdate: async (_, { input }, { base4rest, basedb }, info) => {
+      validateRest(base4rest);
```
Comment (zarathustra323) [inline services/graphql-server/src/graphql/resolvers/platform/asset.js]: Should require auth

File: services/graphql-server/src/graphql/resolvers/platform/asset.js [hunk 3]
```diff
@@ -14,4 +18,24 @@ module.exports = {
...
+      const type = 'platform/asset/image';
+      const { id, ...payload } = input;
+      const keys = Object.keys(payload);
+      if (!keys.length) throw new UserInputError('You must specify a field to update!');
```
Comment (zarathustra323) [inline services/graphql-server/src/graphql/resolvers/platform/asset.js]: No need to throw an error.

File: services/graphql-server/src/graphql/resolvers/platform/asset.js (hunk 2 above)
Comment (solocommand) [reply services/graphql-server/src/graphql/resolvers/platform/asset.js]: The auth requirement is done in the directive, doesn't need to do anything on the resolver side -- but I did forget to make the definition use the directive, so i will update that ;P

File: services/graphql-server/src/graphql/resolvers/platform/asset.js [hunk 4]
```diff
@@ -14,4 +17,23 @@ module.exports = {
...
   },
+
+  /**
+   *
+   */
+  Mutation: {
+    updateAssetImage: async (_, { input }, { base4rest, basedb }, info) => {
+      validateRest(base4rest);
+      const type = 'platform/asset/image';
+      const { id, ...payload } = input;
```
Comment (zarathustra323) [inline services/graphql-server/src/graphql/resolvers/platform/asset.js]: Where is the `...payload` variable? I don't see it.

File: services/graphql-server/src/graphql/definitions/platform/asset.js [hunk 5]
```diff
@@ -71,6 +75,14 @@ input AssetImageQueryInput {
   id: ObjectID!
 }
 
+input AssetImageMutationInput {
+  id: ObjectID!
+  name: String
```
Comment (zarathustra323) [inline services/graphql-server/src/graphql/definitions/platform/asset.js]: You should probably move these fields to a `payload` child-input object (see below)

File: services/graphql-server/src/graphql/definitions/platform/asset.js [hunk 6]
```diff
@@ -75,6 +76,18 @@ input AssetImageQueryInput {
   id: ObjectID!
 }
 
+input UpdateAssetImageMutationInput {
+  id: ObjectID!
+  payload: UpdateAssetImageMutationPayloadInput!
```
Comment (zarathustra323) [inline services/graphql-server/src/graphql/definitions/platform/asset.js]: `payload: UpdateAssetImageMutationPayloadInput = {}`

File: services/graphql-server/src/graphql/utils/project-find-one.js [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+module.exports = ({
+  queryInfo,
+  basedb,
+  type,
+  model,
+  id,
+} = {}) => {
+  const { fieldNodes, schema, fragments } = queryInfo;
+  const projection = getProjection(
+    schema,
```
Comment (zarathustra323) [inline services/graphql-server/src/graphql/utils/project-find-one.js]: What is findOne "actually" doing here?

**Pull request merged**